File: triangles.c
#include <stdio.h>
#include <math.h>
#include <stdlib.h>
int main (){

	double X1, Y1, 
	       X2, Y2,
	       X3, Y3;
	double a, b, c;

	double X4, Y4, 
	       X5, Y5,
	       X6, Y6;
	double d, e, f;

	double P1, P2;



	printf("Triangle #1:\n");
	printf("Vertex A:\n");
	if (scanf("%lf %lf", &X1, &Y1) !=2){
	printf("Invalid input.\n");
		return 0;}
	
	printf("Vertex B:\n");
	if (scanf("%lf %lf", &X2, &Y2) !=2){
	printf("Invalid input.\n");
		return 0;}

	printf("Vertex C:\n");
	if (scanf("%lf %lf", &X3, &Y3) !=2){
	printf("Invalid input.\n");
		return 0;}

	a = sqrt((X1-X2)*(X1-X2)+(Y1-Y2)*(Y1-Y2));
	b = sqrt((X1-X3)*(X1-X3)+(Y1-Y3)*(Y1-Y3));
	c = sqrt((X2-X3)*(X2-X3)+(Y2-Y3)*(Y2-Y3));



	if   (a+b-c<= 1e-12 *c*c || a+c-b<= 1e-12*b*b || b+c-a<= 1e-12*a*a){
	printf("The points do not form a triangle.\n");
	exit(1);
	}





	printf("Triangle #2:\n");
	printf("Vertex A:\n");
	if (scanf("%lf %lf", &X4, &Y4) !=2){
	printf("Invalid input.\n");
		return 0;}
	
	printf("Vertex B:\n");
	if (scanf("%lf %lf", &X5, &Y5) !=2){
	printf("Invalid input.\n");
		return 0;}

	printf("Vertex C:\n");
	if (scanf("%lf %lf", &X6, &Y6) !=2){
	printf("Invalid input.\n");
		return 0;}




	d = sqrt((X4-X6)*(X4-X6)+(Y4-Y6)*(Y4-Y6));
	e = sqrt((X4-X5)*(X4-X5)+(Y4-Y5)*(Y4-Y5));
	f = sqrt((X5-X6)*(X5-X6)+(Y5-Y6)*(Y5-Y6));

	 P1 = a+b+c;
	 P2 = d+e+f;

	if   (d+e-f<= 1e-12 *f*f || e+f-d<= 1e-12*d*d || d+f-e<= 1e-12*e*e){
	printf("The points do not form a triangle.\n");
	exit(1);
	}
	
	if ((a == d || a == e || a == f) && (b == d || b == e || b ==f) && (c == d || c == e || c == f)) {
	printf("The triangles are identical.\n");
	exit (1);}
	if ((P1 - P2 <= 1e-12 * P2 * P2) && (P2 - P1 <= 1e-12 * P1 * P1))  {
	printf("The triangles are not identical, however, they have the same circumference.\n");
		exit(1);}

	else if (  P1-P2> 1e-12 *P2*P2){
	printf("Triangle #1 has a longer circumference.\n");
	exit(1);}

	else {
	printf("Triangle #2 has a longer circumference.\n");
	exit(1);}




return 0;}

File: triangles.java
package com.company;


import java.util.Scanner;

public class Main {

    public static void main(String[] args) {

        double  x1,y1,
                x2,y2,
                x3, y3,
                x4,y4,
                x5,y5,
                x6,y6;

        double a,b,c,d,e,f;
        double per1, per2;

        Scanner num = new Scanner(System.in);

        System.out.println("Triangle #1:");
        System.out.println("Vertex A:");

if (!num.hasNextDouble()){
    System.out.println("Invalid input!");
    System.exit(1);}
    x1 = num.nextDouble();
        if (!num.hasNextDouble()){
            System.out.println("Invalid input!");
            System.exit(1);}
    y1 = num.nextDouble();

    System.out.println("Vertex B:");
        if (!num.hasNextDouble()){
            System.out.println("Invalid input!");
            System.exit(1);}
    x2 = num.nextDouble();
        if (!num.hasNextDouble()){
            System.out.println("Invalid input!");
            System.exit(1);}
    y2 = num.nextDouble();
    System.out.println("Vertex C:");
        if (!num.hasNextDouble()){
            System.out.println("Invalid input!");
            System.exit(1);}
    x3 = num.nextDouble();
        if (!num.hasNextDouble()){
            System.out.println("Invalid input!");
            System.exit(1);}
    y3 = num.nextDouble();

      a = Math.sqrt( (x1-x2)*(x1-x2) + (y1-y2)*(y1-y2) );
      b = Math.sqrt( (x1-x3)*(x1-x3) + (y1-y3)*(y1-y3) );
      c = Math.sqrt( (x2-x3)*(x2-x3) + (y2-y3)*(y2-y3));

        if   (a+b-c<= 1e-12 *c*c || a+c-b<= 1e-12*b*b || b+c-a<= 1e-12*a*a){
            System.out.println("The points do not form a triangle");
            System.exit(1);
        }
        System.out.println("Triangle #2:");
        System.out.println("Vertex A:");
        if (!num.hasNextDouble()){
            System.out.println("Invalid input!");
            System.exit(1);}
        x4 = num.nextDouble();
        if (!num.hasNextDouble()){
            System.out.println("Invalid input!");
            System.exit(1);}
        y4 = num.nextDouble();


        System.out.println("Vertex B:");
        if (!num.hasNextDouble()){
            System.out.println("Invalid input!");
            System.exit(1);}
        x5 = num.nextDouble();
        if (!num.hasNextDouble()){
            System.out.println("Invalid input!");
            System.exit(1);}
        y5 = num.nextDouble();

        System.out.println("Vertex C:");
        if (!num.hasNextDouble()){
            System.out.println("Invalid input!");
            System.exit(1);}
        x6 = num.nextDouble();
        if (!num.hasNextDouble()){
            System.out.println("Invalid input!");
            System.exit(1);}
        y6 = num.nextDouble();

        d = Math.sqrt((x4-x6)*(x4-x6)+(y4-y6)*(y4-y6));
        e = Math.sqrt((x4-x5)*(x4-x5)+(y4-y5)*(y4-y5));
        f = Math.sqrt((x5-x6)*(x5-x6)+(y5-y6)*(y5-y6));

        per1 = a+b+c;
        per2 = d+e+f;

        if   (d+e-f<= 1e-12 *f*f || e+f-d<= 1e-12*d*d || d+f-e<= 1e-12*e*e){
            System.out.println("The points do not form a triangle.");
        System.exit(1);
        }

        if ((a == d || a == e || a == f) && (b == d || b == e || b ==f) && (c == d || c == e || c == f)) {
            System.out.println("The triangles are identical.");
           System.exit (0);}
        if ((per1 - per2 <= 1e-12 * per2 * per2) && (per2 - per1 <= 1e-12 * per1 * per1))  {
            System.out.println("The triangles are not identical, however, they have the same circumference.");
            System.exit(0);}

        else if (  per1-per2> 1e-12 *per2*per2){
            System.out.println("Triangle #1 has a longer circumference.");
            System.exit(0);}

        else {
            System.out.println("Triangle #2 has a longer circumference.");
            System.exit(0);
        }

            }

    }

